core-web/libs/edit-content/src/lib/components/dot-edit-content-sidebar/components/dot-edit-content-sidebar-activities/dot-edit-content-sidebar-activities.component.ts

```diff
@@ -1,44 +1,184 @@
 import { CommonModule } from '@angular/common';
-import { ChangeDetectionStrategy, Component, computed, input } from '@angular/core';
-import { ReactiveFormsModule } from '@angular/forms';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    effect,
+    inject,
+    input,
+    output
+} from '@angular/core';
+import {
+    FormBuilder,
+    FormControl,
+    FormGroup,
+    ReactiveFormsModule,
+    Validators
+} from '@angular/forms';
 
 import { AvatarModule } from 'primeng/avatar';
 import { ButtonModule } from 'primeng/button';
+import { DataViewModule } from 'primeng/dataview';
 import { InputTextareaModule } from 'primeng/inputtextarea';
 import { SkeletonModule } from 'primeng/skeleton';
 
 import { ComponentStatus } from '@dotcms/dotcms-models';
-import { DotGravatarDirective, DotMessagePipe, DotRelativeDatePipe } from '@dotcms/ui';
+import {
+    DotFieldValidationMessageComponent,
+    DotGravatarDirective,
+    DotMessagePipe,
+    DotRelativeDatePipe
+} from '@dotcms/ui';
 
-import { Activity } from '../../../../models/dot-edit-content.model';
+import { Activity, DotContentletState } from '../../../../models/dot-edit-content.model';
 import { DotEditContentSidebarActivitiesSkeletonComponent } from '../dot-edit-content-sidebar-activities-skeleton/dot-edit-content-sidebar-activities-skeleton.component';
 
+const COMMENT_MIN_LENGTH = 3;
+const COMMENT_MAX_LENGTH = 500;
+
 /**
- * Component that displays a list of activities in the content sidebar
+ * Component that displays and manages activities in the content sidebar.
+ * Allows users to view activity history and add new comments.
  */
 @Component({
     selector: 'dot-edit-content-sidebar-activities',
     standalone: true,
     imports: [
         CommonModule,
         ReactiveFormsModule,
+        AvatarModule,
         ButtonModule,
+        DataViewModule,
         InputTextareaModule,
         DotMessagePipe,
         SkeletonModule,
-        AvatarModule,
+        DotGravatarDirective,
         DotRelativeDatePipe,
         DotEditContentSidebarActivitiesSkeletonComponent,
-        DotGravatarDirective
+        DotFieldValidationMessageComponent
     ],
     templateUrl: './dot-edit-content-sidebar-activities.component.html',
     styleUrls: ['./dot-edit-content-sidebar-activities.component.scss'],
-
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class DotEditContentSidebarActivitiesComponent {
+    /**
+     * Form builder for the comment field
+     */
```

Comment:
In this case, I think the this type definition `FormGroup<{ comment: FormControl<string> }` is unnecessary, because  `this.#fb.group` already returns the proper typing.